dbms/src/Storages/S3/S3Common.cpp

```diff
@@ -712,17 +718,9 @@ void ensureLifecycleRuleExist(const TiFlashS3Client & client, Int32 expire_days)
     if (lifecycle_rule_has_been_set)
     {
         LOG_INFO(client.log, "The lifecycle rule has been set, n_rules={} filter={}", old_rules.size(), TaggingObjectIsDeleted);
-        return;
-    }
-    else
-    {
-        UNUSED(expire_days);
-        LOG_WARNING(client.log, "The lifecycle rule with filter \"{}\" has not been set, please check the bucket lifecycle configuration", TaggingObjectIsDeleted);
-        return;
+        return true;
     }
 
-#if 0
-    // Adding rule by AWS SDK is failed, don't know why
```

Comment:
This PR each time will try and log a warnings each time S3GC is run. If the check fail, just log a warning and continue the S3 GC process.
Adding rule is still rejected by AWS, will find out why in later PRs.